index.d.ts

```diff
@@ -126,21 +126,21 @@ export interface RetryOptions {
 
 	@default ['get', 'put', 'head', 'delete', 'options', 'trace']
 	*/
-	methods?: string[];
+	methods?: string[] | Set<string>;
```

Comment:
The tsd test fails without this change. We need to allow `retry.methods` to be a `Set` if we want to support the use case from #167. Otherwise, when Ky nornalizes it and passes it to hooks, the hooks won't be able to call Ky with the options they received.

Reply:
I meant getting rid of `Set` here:

https://github.com/sindresorhus/ky/blob/c5821a7fbf1af21f934ee5ed5a76fb224559c530/index.js#L201-L202

TBH I'm good with both, just that `Set` won't affect speed in a big way.

Reply:
I see. Right, that's true. We don't _have_ to normalize it to a `Set`. It's just that a `Set` is slightly more "correct".

I'm fine with it either way, too. There's no meaningful difference in terms of complexity or performance.

Reply:
Yeah, let’s do Array for simplicity.

Reply:
Done. Dropped support for `Set`s throughout the API, to keep it consistent.